Add azure-kit-demo module

heruan opened the pull request:
This adds the `azure-kit-demo` module that includes a basic Start Vaadin app to be used as a demo and for testing kit features.

The module contains Docker and Kubernetes descriptors to be ready to be deployed on a cluster.

mcollovati commented on azure-kit-demo/kubernetes.yaml:
```diff
@@ -0,0 +1,40 @@
+kind: Deployment
+metadata:
+  name: azure-kit-demo
+  namespace: default
```
I would remove the namespace from the deployment definition, so it can be defined when applying the configuration

heruan replied:
Good point, done!

mcollovati commented on azure-kit-demo/src/main/java/com/vaadin/azure/demo/Application.java:
```diff
@@ -0,0 +1,29 @@
+@PWA(name = "Azure Kit Demo", shortName = "Azure Kit Demo", offlineResources = {})
+@NpmPackage(value = "line-awesome", version = "1.3.0")
+@NpmPackage(value = "@vaadin-component-factory/vcf-nav", version = "1.0.6")
+public class Application implements AppShellConfigurator {
```
What about adding `@Push` annotation to be able to check that web socket works seamlessly on the application deployed on the cluster?

mcollovati commented:
Can we add a `.gitignore` in the demo module to exclude all build related files?

I would also add a `Dockerfile` to build image against local built artifact, since the actual `Dockerfile` fails on first stage because it does not find the parent pom.

```
FROM openjdk:17-jdk-slim
COPY target/*.jar /usr/app/app.jar
RUN useradd -m myuser
USER myuser
EXPOSE 8080
CMD java -jar /usr/app/app.jar
```

heruan commented:
> Can we add a `.gitignore` in the demo module to exclude all build related files?

Sure, added!
 
> I would also add a `Dockerfile` to build image against local built artifact, since the actual `Dockerfile` fails on first stage because it does not find the parent pom.

True, I've modified the file as suggested and also added a README with instructions to run the demo.

mcollovati commented on azure-kit-demo/README.md:
```diff
@@ -0,0 +1,55 @@
+At http://localhost:8000/counter you find a counter which value is held in the UI.
+Pushing the "Increment" button increments the counter and keeps a log of the operation, tracking the hostname and address of the node currently serving the request.
+
+Now try to scale down your deploymen to a single pod:
```
```suggestion
Now try to scale down your deployment to a single pod:
```

heruan replied:
Done.

mcollovati commented on azure-kit-demo/README.md:
````diff
@@ -0,0 +1,55 @@
+
+1. Build the application JAR:
+```
+# mvn clean package -Pproduction -Dmaven.javadoc.skip=true -Dmaven.source.skip=true
````
System properties can now be removed, since they are configured in POM file
```suggestion
# mvn clean package -Pproduction
```

heruan replied:
Done.
